### PR TITLE
feat(cli): add obfuscate flag for diagnostics proxy-metrics cmd

### LINKED ISSUE #6073: diagnostics proxy-metrics --obfuscate
The `proxy-metrics` diagnostics include potentially private information (service names, pod names, etc).

It would be helpful if this command could support an `--obfuscate` flag that mangles metric label values like

- `authority`
- `dst_service`
- `dst_namespace`
- `target_addr`
- `server_id`
- `client_id`

Each of these values should be hashed so that values can be correlated but should be obscured so that the metrics can be shared publicly.

### PR DESCRIPTION
**Pull request by ahmedalhulaibi**
The proxy-metrics diagnostics include potentially private information (service names, pod names, etc).

This commit adds an obfuscate flag to the diagnostics proxy-metrics command to obfuscate this data

`diagnostics proxy-metrics --obfuscate`

Closes #6073

Signed-off-by: ahmedalhulaibi <ahmed.alhulaibi41@gmail.com>

### CODE REVIEW

File: cli/cmd/metrics_diagnostics_util.go [hunk 1]
```diff
@@ -125,12 +128,13 @@ func getMetrics(
...
 	}
 
+wait:
 	for {
 		select {
 		case result := <-resultChan:
 			results = append(results, result)
 		case <-time.After(waitingTime):
-			break // timed out
+			break wait // timed out
```
Comment (kleimkuhler) [inline cli/cmd/metrics_diagnostics_util.go:137]: Hm not sure how we haven't caught this before. Thanks!

File: cli/cmd/metrics.go [hunk 2]
```diff
@@ -85,11 +87,20 @@ func newCmdMetrics() *cobra.Command {
...
+				case result.err != nil:
+					content += fmt.Sprintf("# ERROR: %s\n", result.err)
+				case result.err == nil && options.obfuscate:
+					obfuscatedMetrics, err := obfuscateMetrics(result.metrics)
+					if err != nil {
+						content += fmt.Sprintf("# ERROR %s\n", err)
+					} else {
+						content += string(obfuscatedMetrics)
+					}
+				case result.err == nil && !options.obfuscate:
```
Comment (kleimkuhler) [inline cli/cmd/metrics.go:100]: We should be able to simplify this `switch` a little

```suggestion
				switch {
				case result.err != nil:
					content += fmt.Sprintf("# ERROR: %s\n", result.err)
				case options.obfuscate:
					obfuscatedMetrics, err := obfuscateMetrics(result.metrics)
					if err != nil {
						content += fmt.Sprintf("# ERROR %s\n", err)
					} else {
						content += string(obfuscatedMetrics)
					}
				default:
					content += string(result.metrics)
				}
```

**Review by kleimkuhler — COMMENTED** [review]
Overall this looks good and works well for me! Left a trivial comment about simplifying the `switch` statement.

**Review by adleong — COMMENTED** [review]
This works great for me as well!

The hashes are a bit long and unwieldy, e.g. 

```
response_latency_ms_bucket{direction="inbound",target_addr="f21fd97771d00f42464ae5967f97496c36e2f815cca7d919b396ae3007d2cfc0",target_ip="0.0.0.0",target_port="4191",tls="no_identity",no_tls_reason="no_tls_from_remote",srv_name="default:all-unauthenticated",saz_name="default:all-unauthenticated",status_code="200",le="3"} 597
```

What do you think about truncating the hashes to something shorter?  Perhaps to 7 characters like git's short hashes?

File: cli/cmd/metrics.go (hunk 2 above)
Comment (ahmedalhulaibi) [reply cli/cmd/metrics.go]: Ah yes thanks for catching this, I implemented the suggestion in fb44bef

**Comment by ahmedalhulaibi** [comment]
@adleong That's a fair point, it was certainly difficult to read on my terminal. I adjusted the output to 8 characters in fb44bef

File: cli/cmd/metrics_diagnostics_util.go (hunk 1 above)
Comment (ahmedalhulaibi) [reply cli/cmd/metrics_diagnostics_util.go:137]: I expected the linter to pick it up, but upon inspection the `staticcheck` linter is explicitly disabled for this repo.

https://github.com/linkerd/linkerd2/blob/c031939ea4af873ec85a8c00a651081f7d915de6/.golangci.yml#L43-L46

**Review by kleimkuhler — APPROVED** [review]
Thanks!

**Review by adleong — APPROVED** [review]
Nice!

**Pull request merged**